TaskServiceInterview.cs

```diff
@@ -0,0 +1,448 @@
+            IRepository<Classification> classificationRepository,
+            IAuditLogServant auditLogServant,
+            ITaskLockServant taskLockServant,
+            ITaskChangeNotificationServant taskChangeNotificationServant,
+            IClock clock,
+            IPermissionServant permissionServant,
+            IClassificationAccessServant classificationAccessServant,
+            IValidator<TaskEntity> taskValidator,
+            IAsynchronousOperator asynchronousOperator,
+            ICustomerDetailsServant customerDetailsServant,
+            IBusinessCalendarClient businessCalendarClient,
+            IPsbConnector psbConnector,
+            INotificationService notificationService,
+            ITaskBusinessEventFinder taskBusinessEventFinder,
+            IInternalReminderFinder internalReminderFinder,
+            ITaskCreateService taskCreateService,
+            IRepository<TaskConnection> taskConnectionRepository)
+        {
+            this.notificationService = notificationService;
+            this.taskBusinessEventFinder = taskBusinessEventFinder;
+            this.internalReminderFinder = internalReminderFinder;
+            this.taskCreateService = taskCreateService;
+            this.taskFinder = taskFinder;
+            this.classificationRepository = classificationRepository;
+            this.mappingServant = mappingServant;
+            this.repository = repository;
+            this.auditLogServant = auditLogServant;
+            this.taskLockServant = taskLockServant;
+            this.taskChangeNotificationServant = taskChangeNotificationServant;
+            this.clock = clock;
+            this.classificationRepository = classificationRepository;
+            this.permissionServant = permissionServant;
+            this.classificationAccessServant = classificationAccessServant;
+            this.taskValidator = taskValidator;
+            this.asynchronousOperator = asynchronousOperator;
+            this.businessCalendarClient = businessCalendarClient;
+            this.customerDetailsServant = customerDetailsServant;
+            this.psbConnector = psbConnector;
+            this.taskConnectionRepository = taskConnectionRepository;
+        }
+
+        public TaskDetailsItem GetById(Guid id, string employeeInitials)
+        {
+            var task = this.repository.Single(id, t => t.FlexType,
+                t => t.Classification, t => t.ExternalParty, t => t.Metadata);
+
+            if (task.Status != TaskStatus.Finished && task.IsHighPriority == false && task.Deadline > this.clock.Current && task.AssigneeInitials == null &&
+
+                (int)(task.RequiredPermission & PermissionType.PensionStandard)  == 2 &&
+                this.permissionServant.CanSeeDetails(task, employeeInitials) == false)
+                throw new InsufficientPrivilegesException();
+
+            return this.MapToTaskDetails(task);
+
+        }
+
+        public bool RemoveOwnership(Guid taskId, string userInitials)
```

Comment:
dziwne to zwracanie booli przy metodach, ktore nic nie musza zwracac. Jezeli juz to IsOwnershipRemoved